ultra/ultra/evaluate.py

```diff
@@ -58,24 +54,37 @@ def evaluation_check(
     headless,
     log_dir,
 ):
-    agent_itr = episode.get_itr(agent_id)
+    agent_iterations = {agent_id: episode.get_itr(agent_id) for agent_id in agent_ids}
+    lowest_agent_iteration = min(agent_iterations.values())
```

Comment:
I think this can be simplified when we change to episode based evaluation. However, why is lowest agent iteration used?

Reply:
Good question, yeah this can definitely be simplified when we move to episode-based evaluation. But just in the interim, I thought that it would be best to evaluate once we know all agents have reached the desired eval_rate threshold before we evaluate all the agents together in a test scenario (rather than having just one, or some, agent reach the threshold and then evaluating all the agents in the test scenario). Let me know your thoughts on this or if you have a better idea.